File: src/pratice/pointtopoint/두수의합3273.java
package pratice.pointtopoint;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;

public class 두수의합3273 {
    public static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    public static int N, X;
    public static int[] A;

    public static void main(String[] args) throws IOException {
        input();
        Arrays.sort(A);
        pro();
    }

    private static void pro() {
        int res = 0;
        int L = 0;
        int R = N - 1;
        while(L<R){
            if(A[L] + A[R] < X)L++;
            else if(A[L] + A[R] > X)R--;
            else{
                res++;
                L++;
            }
        }
        System.out.println(res);
    }

    private static void input() throws IOException {
        N = Integer.parseInt(br.readLine());
        A = new int[N];
        String[] split = br.readLine().split(" ");
        for (int i = 0; i < N; i++) {
            A[i] = Integer.parseInt(split[i]);
        }
        X = Integer.parseInt(br.readLine());
    }
}

File: src/pratice/pointtopoint/두수의합3273.kt
package pratice.pointtopoint

fun main() {
    val N = readln().toInt()
    val arr = readln().split(" ").map { it.toInt() }.toIntArray()
    val X = readln().toInt()

    arr.sort()
    var L = 0
    var R = N - 1
    var cnt = 0
    while (L < R) {
        if (arr[L] + arr[R] < X)L++
        else if (arr[L] + arr[R] > X)R--
        else {
            cnt++
            L++
        }
    }
    println(cnt)
}
